nbgrader/plugins/export.py

```diff
@@ -56,46 +79,50 @@ def export(self, gradebook):
 
         # Loop over each assignment in the database
         for assignment in gradebook.assignments:
-
-            # Loop over each student in the database
-            for student in gradebook.students:
-
-                # Create a dictionary that will store information about this
-                # student's submitted assignment
-                score = {}
-                score['assignment'] = assignment.name
-                score['duedate'] = assignment.duedate
-                score['student_id'] = student.id
-                score['last_name'] = student.last_name
-                score['first_name'] = student.first_name
-                score['email'] = student.email
-                score['max_score'] = assignment.max_score
-
-                # Try to find the submission in the database. If it doesn't
-                # exist, the `MissingEntry` exception will be raised, which
-                # means the student didn't submit anything, so we assign them a
-                # score of zero.
-                try:
-                    submission = gradebook.find_submission(
-                        assignment.name, student.id)
-                except MissingEntry:
-                    score['timestamp'] = ''
-                    score['raw_score'] = 0.0
-                    score['late_submission_penalty'] = 0.0
-                    score['score'] = 0.0
-                else:
-                    penalty = submission.late_submission_penalty
-                    score['timestamp'] = submission.timestamp
-                    score['raw_score'] = submission.score
-                    score['late_submission_penalty'] = penalty
-                    score['score'] = max(0.0, submission.score - penalty)
-
-                for key in score:
-                    if score[key] is None:
-                        score[key] = ''
-                    if not isinstance(score[key], str):
-                        score[key] = str(score[key])
-
-                fh.write(fmt.format(**score))
+            # test if assignment is requested
+            if not allassignments or assignment.name in allassignments:
```

Comment:
So that four levels of indentation aren't required, I'd suggest the following instead:

```
if allassignments and assignment.name not in allassignments:
    continue

for student in gradebook.students:
    if allstudents and student.id not in allstudents:
        continue

    ...
```